frame/babe/src/lib.rs

```diff
@@ -498,30 +504,38 @@ impl<T: Config> Module<T> {
 	}
 
 	/// Produces information about the current epoch.
-	pub fn current_epoch() -> Epoch {
-		Epoch {
+	pub fn current_epoch() -> StandaloneEpoch {
+		let epoch_config = EpochConfig::get().unwrap();
+
+		StandaloneEpoch {
 			epoch_index: EpochIndex::get(),
 			start_slot: Self::current_epoch_start(),
 			duration: T::EpochDuration::get(),
 			authorities: Self::authorities(),
 			randomness: Self::randomness(),
+			c: epoch_config.c,
+			allowed_slots: epoch_config.allowed_slots,
 		}
 	}
 
 	/// Produces information about the next epoch (which was already previously
 	/// announced).
-	pub fn next_epoch() -> Epoch {
+	pub fn next_epoch() -> StandaloneEpoch {
 		let next_epoch_index = EpochIndex::get().checked_add(1).expect(
 			"epoch index is u64; it is always only incremented by one; \
 			 if u64 is not enough we should crash for safety; qed.",
 		);
 
-		Epoch {
+		let epoch_config = EpochConfig::get().unwrap();
```

Comment:
You can't unwrap / panic in the runtime.

You need to handle the situation where the value does not exist in storage. A Default value makes sense here probably.